packages/jellyfish-transaction/__tests__/script/dftx/dftx_masternode/CreateMasternode.test.ts

```diff
@@ -0,0 +1,59 @@
+import { SmartBuffer } from 'smart-buffer'
+import { OP_DEFI_TX } from '../../../../src/script/dftx'
+import { OP_CODES } from '../../../../src'
+import { toBuffer, toOPCodes } from '../../../../src/script/_buffer'
+import { CCreateMasternode, CreateMasternode } from '../../../../src/script/dftx/dftx_masternode'
+
+it('should bi-directional buffer-object-buffer', () => {
+  const fixtures = [
+    '6a1c44665478430147bfb0a67b85a1718381558434fbfe7c4866cf2e0000', // p2pkh
+    '6a1c4466547843040e12cde53c156560faa1d01d144d234a74b653950000', // p2wpkh
+    '6a1c4466547843040e12cde53c156560faa1d01d144d234a74b653950401', // p2wpkh + timelock 5 years (260)
+    '6a1c4466547843040e12cde53c156560faa1d01d144d234a74b653950802' // p2wpkh + timelock 10 years (520)
+  ]
+
+  fixtures.forEach(hex => {
+    const stack = toOPCodes(
+      SmartBuffer.fromBuffer(Buffer.from(hex, 'hex'))
+    )
+    const buffer = toBuffer(stack)
+    expect(buffer.toString('hex')).toStrictEqual(hex)
+    expect((stack[1] as OP_DEFI_TX).tx.type).toStrictEqual(0x43)
+  })
+})
+
+describe('CreateMasternode', () => {
+  const header = '6a1c4466547843' // OP_RETURN(0x1a) (length 28 = 0x1c) CDfTx.SIGNATURE(0x44665478) CCreateMasternode.OP_CODE(0x43)
+  const data = '01742b337e0f40d5f229a89d3a26d53ae1093b6cff0802' // CreateMasternode.operatorType(0x01) CreateMasternode.operatorAuthAddress(0x742b337e0f40d5f229a89d3a26d53ae1093b6cff) CreateMasternode.timelock(0x0802)
+  const createMasterNode: CreateMasternode = {
+    operatorType: 0x01,
+    operatorAuthAddress: '742b337e0f40d5f229a89d3a26d53ae1093b6cff',
+    timelock: 0x0208
+  }
+
+  it('should craft dftx with OP_CODES._()', () => {
+    const stack = [
+      OP_CODES.OP_RETURN,
+      OP_CODES.OP_DEFI_TX_CREATE_MASTER_NODE(createMasterNode)
+    ]
+
+    const buffer = toBuffer(stack)
+    expect(buffer.toString('hex')).toStrictEqual(header + data)
+  })
+
+  describe('Composable', () => {
+    it('should compose from buffer to composable', () => {
+      const buffer = SmartBuffer.fromBuffer(Buffer.from(data, 'hex'))
+      const composable = new CCreateMasternode(buffer)
+      expect(composable.toObject()).toStrictEqual(createMasterNode)
+    })
+
+    it('should compose from composable to buffer', () => {
+      const composable = new CCreateMasternode(createMasterNode)
+      const buffer = new SmartBuffer()
+      composable.toBuffer(buffer)
+
+      expect(buffer.toBuffer().toString('hex')).toStrictEqual(data)
+    })
+  })
+})
```

Comment:
Need tests here for sending create masternode tx, also test with timelock, no timelock etc

Reply:
huh?? timelock is mandatory now in rawtx section 
sending createmn tx test is here
https://github.com/DeFiCh/jellyfish/blob/e224ab4626b7e4ee186f6e5188789c6f3888bd7e/packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_masternode_create.test.ts

but timelock in rpc call is optional

Reply:
thanks I see that file now! I looked for ages and couldn't find it at first, I was looking at the composer and the txn-builder

for no timelock, I request a test to decompose pre Eunos Paya txns

Reply:
hmm.. sounds like my previous commit should be correct.. the preEunosPaya doesn't have timelock impl.. the timelock should not be mandatory in rawtx section...